internal/abi/fastly/hostcalls_guest.go

```diff
@@ -3387,3 +3387,55 @@ func PurgeSurrogateKey(surrogateKey string, opts PurgeOptions) error {
 		prim.ToPointer(&opts.opts),
 	).toError()
 }
+
+// witx:
+//
+//	(module $fastly_device_detection
+//	    (@interface func (export "lookup")
+//	        (param $user_agent string)
+//
+//	        (param $buf (@witx pointer (@witx char8)))
+//	        (param $buf_len (@witx usize))
+//	        (param $nwritten_out (@witx pointer (@witx usize)))
+//	        (result $err (expected (error $fastly_status)))
+//	    )
+//	)
+//
+//go:wasmimport fastly_device_detection lookup
+//go:noescape
+func fastlyDeviceDetectionLookup(
+	userAgentData prim.Pointer[prim.U8], userAgentLen prim.Usize,
+	buf prim.Pointer[prim.Char8],
+	bufLen prim.Usize,
+	nWritten prim.Pointer[prim.Usize],
+) FastlyStatus
+
+func DeviceLookup(userAgent string) ([]byte, error) {
+	buf := prim.NewWriteBuffer(defaultBufferLen)
```

Comment:
Is the default buffer length the same in other SDKs?

Reply:
In our SDK it's 16k.  In Rust it's only 1k.  The Rust SDK is better about using "adaptive buffers", in which you make a hostcall with a certain sized buffer and if it's too small the SDK tells you the size it needs, and you allocate a new buffer to fit it.  We do it for a few hostcalls in the Go SDK (and any new one I write, including this one) but it isn't pervasive.

I can resize this one down to 1k to match it.

Reply:
Thank you for the explanation. I think it's fine. I was more concerned about being smaller, in which cases we would need to do 2 calls. I think we can leave it as is.